core/src/main/java/com/scalar/db/transaction/consensuscommit/TransactionalTableMetadataManager.java

```diff
@@ -0,0 +1,87 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+
+public class TransactionalTableMetadataManager {
```

Comment:
This class retrieves a `TableMetadata` instance from `DistributedStorageAdmin` and convert it to a `TransactionalTableMetadata` instance and cache it.